add terraform-airbyte-provider page, images, and ref for nav.adoc (#172)
Teradata/quickstarts: modules/elt/pages/terraform-airbyte-provider.adoc

> = Automate ELT workflows with Terraform and Airbyte on Teradata Vantage 

Reviewer: ```suggestion
= Manage ELT pipelines as code with Terraform and Airbyte on Teradata Vantage 
```

:experimental:
:page-author: Janeth Graziani
:page-email: Janeth.graziani@teradata.com
:page-revdate: February 24, 2024

> :description: Use Airbyte's Terraform provider to automate and version-control your Airbyte ELT workflows on Teradata Vantage.

Reviewer: ```suggestion
:description: Use Terraform to manage Teradata data pipelines in Airbyte using Terraform.
```

:keywords: Terraform, Airbyte, Teradata Vantage, data engineering, ELT, automation, data integration, CI/CD, version control  
:dir: terraform-airbyte-provider

== Overview 

> In this Quickstart, you'll learn to use Terraform to automate Airbyte ELT workflows on Teradata Vantage. Instead of manual configurations through the WebUI, we'll use code to manage Airbyte resources. The provided example illustrates a basic ELT workflow from Google Sheets to Teradata Vantage using Airbyte's Terraform provider.

Reviewer: We should keep a more impersonal tone

```suggestion
This quickstart explains how to use Terraform to manage Airbyte data pipelines as code. Instead of manual configurations through the WebUI, we'll use code to create and configure Airbyte resources. The provided example illustrates a basic ELT pipeline from Google Sheets to Teradata Vantage using Airbyte's Terraform provider.
```

== Install Terraform 
* Apply the respective commands to install Terraform on your Operating system. Find additional options on the link:https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli[Terraform site].

[tabs, id="terraform_tab_install", role="emits-gtm-events"]
====

macOS::
+
First, install the HashiCorp tap, a repository of all https://brew.sh/[Homebrew] packages.
+
[source,bash]
----
 brew tap hashicorp/tap
----
+
Next, install Terraform with hashicorp/tap/terraform.
+ 
[source,bash]
----
 brew install hashicorp/tap/terraform
----
Windows::
+
https://chocolatey.org/[Chocolatey] is a free and open-source package management system for Windows. Install the Terraform package from the command-line.
+
[source,powershell]
----
 choco install terraform
----
Linux::
+ 
[source,bash]
----
wget -O- https://apt.releases.hashicorp.com/gpg | sudo gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg
echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/hashicorp.list
sudo apt update && sudo apt install terraform 
----
====

> == Environment Preparation

Reviewer: I've not written it anywhere yet, but we don't capitalize section titles :D

```suggestion
== Environment preparation
```

Prepare your environment by creating a directory for your Terraform configuration and initialize two files: `main.tf` and `variable.tf`.

[source, bash]
----
mkdir terraform_airbyte
cd terraform_airbyte
touch main.tf variables.tf
----

> == Define your Data Workflow

Reviewer: ```suggestion
== Define a data pipeline
```

Connection: # airbyte_connection.googlesheets_teradata will be created

Destination: # airbyte_connection.googlesheets_teradata will be created

Source: # airbyte_source_google_sheets.my_source_gsheets will be created
  
image::{dir}/terraformplan.png[View Terraform execution plan with terraform plan command, align="center" width=75%]

Run `terraform apply` and `yes` to generate a plan and carry out the plan.

image::{dir}/terraformapply.png[Apply the Terraform plan with terraform apply command, align="center" width=75%]

The `terraform.tfstate` file is created after you run `terraform apply` for the first time. This file tracks the status of all sources, destinations, and connections managed by Terraform. For subsequent executions of `Terraform apply`, Terraform compares the code in the `main.tf` file with the code stored in the `tfstate` file. If resources are added or removed in `main.tf`, Terraform automatically updates both your deployment and the `.tfstate` file accordingly upon deployment. Do not modify this file by hand.

You now have a Source, Destination and Connection on Airbyte Cloud created and managed via Terraform. 

> image::{dir}/airbyteconnection.png[Airbyte Connection in Airbyte Cloud UI, align="center" width=75%]

Reviewer: Are there other resources we can recommend, like the 2 other airbyte-related quickstarts?
